### PR TITLE
Bump v8 peer deps to support react 18

### LINKED ISSUE #24975: Bump react peer-deps to support React 18

### PR DESCRIPTION
**Pull request by micahgodbolt**
fixes #24975

To establish confidence that react will function properly, we have:

1. triaged all known issues for using react in React 18
2. created a test app (react-test-tests-v8) to test issues we've already fixed, and provide a platform for future triage and testing
3. Publishing a custom [storybook release](https://628d031b55e942004ac95df1-awmxweszfs.chromatic.com/?path=/story/activityitem--root) with current controls, rendered in React 18 + strict mode

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/adea89e6cc51161fbb544f465e298f60046fd55c'>adea89e6cc51161fbb544f465e298f60046fd55c</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/BaseButton-mount_base.html">1173</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/BaseButton-mount.html">1177</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Breadcrumb-mount_base.html">3001</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Breadcrumb-mount.html">2948</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Checkbox-mount_base.html">2452</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Checkbox-mount.html">2935</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/CheckboxBase-mount_base.html">2322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/CheckboxBase-mount.html">2359</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ChoiceGroup-mount_base.html">4311</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ChoiceGroup-mount.html">4264</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ComboBox-mount_base.html">1269</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ComboBox-mount.html">1201</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/CommandBar-mount_base.html">9835</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/CommandBar-mount.html">9589</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ContextualMenu-mount_base.html">11611</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ContextualMenu-mount.html">11477</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DefaultButton-mount_base.html">1395</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DefaultButton-mount.html">1500</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DetailsRow-mount_base.html">3656</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DetailsRow-mount.html">3577</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DetailsRowFast-mount_base.html">3651</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DetailsRowFast-mount.html">3596</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DetailsRowNoStyles-mount_base.html">3482</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DetailsRowNoStyles-mount.html">3470</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Dialog-mount_base.html">3333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Dialog-mount.html">3091</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DocumentCardTitle-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/DocumentCardTitle-mount.html">613</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Dropdown-mount_base.html">3237</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Dropdown-mount.html">3232</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/FocusTrapZone-mount_base.html">2005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/FocusTrapZone-mount.html">2027</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/FocusZone-mount_base.html">1990</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/FocusZone-mount.html">1962</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedList-mount_base.html">54899</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedList-mount.html">61138</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedList-virtual-rerender_base.html">25815</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedList-virtual-rerender.html">25164</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">95395</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedList-virtual-rerender-with-unmount.html">94362</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedListV2-mount_base.html">551</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedListV2-mount.html">556</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedListV2-virtual-rerender_base.html">526</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedListV2-virtual-rerender.html">517</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">544</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">563</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/IconButton-mount_base.html">1916</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/IconButton-mount.html">1896</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Label-mount_base.html">680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Label-mount.html">721</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Layer-mount_base.html">4591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Layer-mount.html">4133</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Link-mount_base.html">843</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Link-mount.html">937</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/MenuButton-mount_base.html">1680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/MenuButton-mount.html">1711</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/MessageBar-mount_base.html">2284</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/MessageBar-mount.html">2217</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Nav-mount_base.html">3305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Nav-mount.html">3262</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/OverflowSet-mount_base.html">1369</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/OverflowSet-mount.html">1373</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Panel-mount_base.html">2566</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Panel-mount.html">2590</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Persona-mount_base.html">1278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Persona-mount.html">1261</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Pivot-mount_base.html">1645</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Pivot-mount.html">1636</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/PrimaryButton-mount_base.html">1532</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/PrimaryButton-mount.html">1520</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Rating-mount_base.html">7043</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Rating-mount.html">7695</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SearchBox-mount_base.html">1506</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SearchBox-mount.html">1517</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Shimmer-mount_base.html">2857</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Shimmer-mount.html">2876</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Slider-mount_base.html">2333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Slider-mount.html">2093</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SpinButton-mount_base.html">4684</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SpinButton-mount.html">4944</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Spinner-mount_base.html">799</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Spinner-mount.html">801</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SplitButton-mount_base.html">3126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SplitButton-mount.html">3109</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Stack-mount_base.html">868</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Stack-mount.html">873</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/StackWithIntrinsicChildren-mount_base.html">2340</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/StackWithIntrinsicChildren-mount.html">2325</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/StackWithTextChildren-mount_base.html">4785</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/StackWithTextChildren-mount.html">4830</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SwatchColorPicker-mount_base.html">10442</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/SwatchColorPicker-mount.html">11122</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/TagPicker-mount_base.html">2638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/TagPicker-mount.html">2669</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/TeachingBubble-mount_base.html">94395</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/TeachingBubble-mount.html">99915</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Text-mount_base.html">787</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Text-mount.html">782</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/TextField-mount_base.html">1613</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/TextField-mount.html">1622</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ThemeProvider-mount_base.html">1456</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ThemeProvider-mount.html">1479</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ThemeProvider-virtual-rerender_base.html">1081</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ThemeProvider-virtual-rerender.html">1035</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2067</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2114</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Toggle-mount_base.html">1090</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/Toggle-mount.html">1038</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/buttonNative-mount_base.html">546</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test/buttonNative-mount.html">558</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/Avatar-mount_base.html">1270</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/Avatar-mount.html">1261</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/Button-mount_base.html">926</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/Button-mount.html">926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProvider-mount_base.html">1493</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProvider-mount.html">1482</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProviderWithTheme-mount_base.html">580</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProviderWithTheme-mount.html">596</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">539</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">545</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">585</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/MakeStyles-mount_base.html">1965</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/MakeStyles-mount.html">1952</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/SpinButton-mount_base.html">2320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24976/perf-test-react-components/SpinButton-mount.html">2346</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-persona/MIGRATION.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ Here's how the API of v8's `Persona` compares to the one from v9's `Persona` com
 - `componentRef` => NOT SUPPORTED - use `ref` instead
 - `hidePersonaDetails` => Use the `Avatar` component for this case
 - ```
+
```
Comment (micahgodbolt) [inline packages/react-components/react-persona/MIGRATION.md:14]: post commit linting added this

File: packages/react-components/react-progress/src/components/Progress/Progress.types.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export type ProgressSlots = {
 export type ProgressProps = Omit<ComponentProps<ProgressSlots>, 'size'> & {
   /**
    * A decimal number between `0` and `1` (or between `0` and `max` if given),
-   * which specifies how much of the task has been completed. 
+   * which specifies how much of the task has been completed.
```
Comment (micahgodbolt) [inline packages/react-components/react-progress/src/components/Progress/Progress.types.ts:20]: post commit linting added this

File: change/@fluentui-react-persona-1132a852-8617-49d5-b3d6-c2afd4597ba6.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "baseline",
```
Comment (micahgodbolt) [inline change/@fluentui-react-persona-1132a852-8617-49d5-b3d6-c2afd4597ba6.json:3]:
```suggestion
  "comment": "prettier update",
```

File: change/@fluentui-react-progress-d2f5b5dc-e2a3-47c1-8a64-38ad980964e3.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "baseline",
```
Comment (micahgodbolt) [inline change/@fluentui-react-progress-d2f5b5dc-e2a3-47c1-8a64-38ad980964e3.json:3]:
```suggestion
  "comment": "prettier update",
```

**Review by khmakoto — COMMENTED** [review]
Should we have any documentation about special things you need to do so the components work correctly in React 18?

**Comment by sopranopillow** [comment]
> 

@micahgodbolt I think the idea is not to have to do anything, right? You should just be able to bump up to R18.

**Review by sopranopillow — APPROVED** [review]

**Review by conniec218 — APPROVED** [review]

**Review by AtishayMsft — APPROVED** [review]

**Comment by micahgodbolt** [comment]
> Should we have any documentation about special things you need to do so the components work correctly in React 18?

As long as you aren't running your React 18 app inside of a yarn v1 mono repo that uses React 17 by default, ya should be good :D

Controls are working out of the box.

**Review by spmonahan — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Pull request merged**